usaspending_api/references/v2/views/filter_trees/tas_filter_tree.py

```diff
@@ -0,0 +1,95 @@
+from usaspending_api.common.helpers.agency_logic_helpers import cfo_presentation_order
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from usaspending_api.common.helpers.agency_logic_helpers import agency_from_identifiers
+from usaspending_api.references.v2.views.filter_trees.filter_tree import DEFAULT_CHILDREN, Node, FilterTree
+from usaspending_api.references.models import ToptierAgency
+
+
+class TASFilterTree(FilterTree):
+    def toptier_search(self):
+        agency_id_sets = TreasuryAppropriationAccount.objects.values("fr_entity_code", "agency_id").distinct(
+            "agency_id"
+        )
+        agency_set = set(
```

Comment:
Is `set()` being used to remove duplicates?  If yes, it likely would be optimal to leverage distinct on the ORM query. If not, perhaps use `list()` to avoid confusion such as mine from future version of ourselves.